packages/xrpl/src/Wallet/index.ts

```diff
@@ -440,7 +440,12 @@ class Wallet {
 
       return memo
     })
-    if (!_.isEqual(decoded, tx)) {
+
+    if (txCopy.TransactionType === 'NFTokenMint' && txCopy.URI) {
+      txCopy.URI = txCopy.URI.toUpperCase()
+    }
```

Comment:
I think this bug should be fixed on the decode side (So that lowercase URIs decode to lowercase URIs again). I think that because URIs can be case sensitive, so our decode should be case sensitive. (Client servers decide how to interpret the URI after the host - https://webmasters.stackexchange.com/questions/5157/is-it-ok-to-have-a-case-insensitive-uri)

Reply:
We require `URI` to be in hex format because it needs to be binary at the protocol level. Here's Rome's explanation for it:

"At the protocol level, it needs to be binary. What form it started out as is immaterial.
So the requirement that it be hex is a detail of the serialization tools. There are two reasons the tools we have now generally require arbitrary binary blob fields to be specified in hex:
It's technologically simpler and more flexible. Our tools can't have bugs in this code because they simply don't have this code at all. :good-thinking:

Automatically encoding/decoding, say, UTF-8 means that you can't also accept hex without having ambiguous cases. ("What if I want to encode a message that happens to also be valid hex?") But if you only accept, say, UTF-8, then there are valid binary values people might want to use that are not valid UTF-8. The URI field is allowed to contain arbitrary data that's not actually a URI; this allows for very small NFTs to be stored actually on-chain. But some of those custom encodings/decodings would be tricky to put in place if you had to write a special serializer to get them into a transaction's binary because they don't produce valid UTF-8.

There are a bunch of potential edge cases in encoding/decoding UTF-8. For example, many UTF-8 characters (many ASCII characters, even) are not valid characters in a URI, but are valid in an IRI (internationalized resource identifier), for example. Should the tool encode these or raise an error?

Requiring hex for binary fields sidesteps these issues and allows each client to choose a way of encoding the data that makes sense for their own circumstances."